virtualizarr/parsers/kerchunk/json.py

```diff
@@ -70,16 +76,25 @@ def __call__(
         content = resp.bytes().to_bytes()
         refs = ujson.loads(content)
 
+        # TODO: Deduplicate with KerchunkParquetParser
         if self.store_registry is None:
             unique_paths = {
                 v[0]
                 for v in refs["refs"].values()
                 if isinstance(v, list) and isinstance(v[0], str)
             }
-            stores = {}
+            registry = ObjectStoreRegistry()
             for path in unique_paths:
-                stores[get_store_prefix(path)] = object_store
-            registry = ObjectStoreRegistry(stores=stores)
+                parsed = urlparse(path)
+                if self.fs_root and not parsed.scheme:
+                    registry.register(f"{self.fs_root}{path}", object_store)
+                elif not parsed.scheme:
+                    raise ValueError(
+                        f"Detected relative path `{path}`. Parser must have `fs_root` set for "
+                        f"handling relative path. Parser has `fs_root = {self.fs_root}`."
+                    )
+                else:
+                    registry.register(path, object_store)
         else:
             registry = self.store_registry
```

Comment:
FYI these changes are not needed if we also adopt https://github.com/zarr-developers/VirtualiZarr/pull/662